core/lib/utils/file_utils.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:developer';
+import 'dart:io';
+import 'package:core/domain/exceptions/download_file_exception.dart';
+import 'package:core/utils/build_utils.dart';
+import 'package:path_provider/path_provider.dart';
+
+class FileUtils {
+
+  static Future<String> _getInternalStorageDirPath(
+    String nameFile,
+    {
+      String? folderPath,
+      String? extensionFile
+    }
+  ) async {
+    if (!BuildUtils.isWeb) {
+
+      String fileDirectory = (await getApplicationDocumentsDirectory()).absolute.path;
```

Comment:
please make sure `getApplicationDocumentsDirectory` is not deprecated in Android version >= 30